chain/cosmos/client.go

```diff
@@ -182,58 +182,81 @@ func (client *Client) SubmitTx(ctx context.Context, tx account.Tx) error {
 
 // AccountNonce returns the current nonce of the account. This is the nonce to
 // be used while building a new transaction.
-func (client *Client) AccountNonce(_ context.Context, addr address.Address) (pack.U256, error) {
+func (client *Client) AccountNonce(ctx context.Context, addr address.Address) (pack.U256, error) {
 	cosmosAddr, err := types.AccAddressFromBech32(string(addr))
 	if err != nil {
 		return pack.U256{}, fmt.Errorf("bad address: '%v': %v", addr, err)
 	}
 
-	accGetter := auth.NewAccountRetriever(client.cliCtx)
-	acc, err := accGetter.GetAccount(Address(cosmosAddr).AccAddress())
-	if err != nil {
-		return pack.U256{}, err
-	}
+	for {
+		select {
+		case <-ctx.Done():
+			return pack.U256{}, ctx.Err()
+		default:
+		}
+
+		accGetter := auth.NewAccountRetriever(client.cliCtx)
+		acc, err := accGetter.GetAccount(Address(cosmosAddr).AccAddress())
+		if err != nil {
```

Comment:
Should probably wait and back-off to avoid spamming.

Reply:
Added a fixed retry timeout. Could add a scaling factor, but I don't think it's necessary.